oecophylla/taxonomy/metaphlan2.py

```diff
@@ -0,0 +1,287 @@
+    A dict, where keys are old IDs and their values are the new merged IDs.
+
+    Raises
+    ------
+    IOError
+        If the file cannot be read.
+    ValueError
+        If IDs of old or merged nodes cannot be converted into int.
+    """
+    return _read_ncbitaxonomy_file(filename)
+
+
+def update_taxids(input, updatedTaxids):
+    """ Updates a map of sequenceIDs to taxIDs with information from merged.dmp
+
+    Some of NCBI's taxonomy IDs might get merged into others. Older sequences
+    can still point to the old taxID. The new taxID must be looked up in the
+    merged.dmp file in order to be able to find the right node in the taxonomy
+    tree.
+
+    Parameters
+    ----------
+    input : dict of dicts of sets
+        The keys of the outer dicts are the sequence types, e.g. NC, GeneID or
+        gi. Keys of the inner dict are OTUs or clades. Values of the inner
+        dict are sets of taxIDs.
+    updatedTaxids : dict
+        Content of merged.dmp in form of a dict where key is current taxID and
+        value the new taxID
+
+    Returns
+    -------
+    The original map, but some taxIDs might have been updated.
+    """
+    for seqType in input:
+        for seqID in input[seqType]:
+            cur_taxid = input[seqType][seqID]
+            if cur_taxid in updatedTaxids:
+                input[seqType][seqID] = updatedTaxids[cur_taxid]
+    return input
+
+
+def generate_map_metaphlan2_ncbitaxids(filename_map, latest_mergeddump=None):
+    """Creates an ID map from metaphlan2 to NCBI taxonomy IDs.
+
+    For every metaphlan2 database, we once need to create a map from their
+    clade names to NCBI taxonomy IDs. Since a clade is a set of several
+    sequences which can stem from different organisms, one clade can point to
+    multiple NCBI taxonomy IDs.
+    The resulting format is one clade per row, tab separated, first column is
+    clade name, second column holds a comma separated list of NCBI taxonomy
+    IDs.
+
+    The file 'Metaphlan/markers_info.txt' is part of metaphlans repository:
+        https://bitbucket.org/biobakery/metaphlan2/src/f19f7cfdc990aa9f2d983
+        7a89bcc5188912c8a82/utils/markers_info.txt.bz2?at=default&fileviewer
+        =file-view-default
+    The file 'metaphlan2_taxids.txt' is the result of a lengthy query against
+    NCBI and can be found here:
+        https://github.com/sjanssen2/ggmap/blob/master/Cache/
+        taxids_metaphlan.txt
+
+    Parameters
+    ----------
+    filename_map : str
+        The filepath for the resulting map from metaphlan2 clades to NCBI
+        taxonomy IDs.
+    latest_mergeddump : str
+        Filepath to "merged.dmp" file of a NCBI taxonomy dump. TaxIDs might
+        have changed between this dump and the generation of the file
+        "metaphlan2_taxids.txt", thus these IDs need to be updated.
+    """
+
+    filename_metaphlan2_marker = "markers_info.txt"
+    if not os.path.exists(filename_metaphlan2_marker):
+        sys.stderr.write(
```

Comment:
Why it writes stderr instead of raise error?

Reply:
what type of error would you suggest for this?

Reply:
Well I am find with it. I just wonder why there is inconsistence thru the code that some errors are raised while other are written to stderr.

Reply:
Here, I think it is more useful to write a clear message to the user rather than annoying him/her with a stack trace

Reply:
Okay cool!